src/tensors/vectors.jl

```diff
@@ -0,0 +1,23 @@
+struct OffByOneVector{T} <: AbstractVector{T}
+    data::AbstractVector{T}
+end
+
+function Base.getindex(vector::OffByOneVector{T}, index::Int) where {T}
```

Comment:
all getters should use `Base.@propagate_inbounds`

Reply:
Done!